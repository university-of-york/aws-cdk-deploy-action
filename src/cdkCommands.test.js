const runCdkCommands = require('./cdkCommands');
const execa = require('execa');

jest.mock('execa');

describe('cdkCommands', () => {
    it('should call execa with correct arguments', async () => {
        await runCdkCommands({
            AWS_ACCOUNT_ID: 'account_id',
            AWS_ROLE_NAME: 'role_name',
            AWS_STACK_NAME: 'stack_name',
            INFRASTRUCTURE_LOCATION: 'some/path/',
        });
        expect(execa.mock.calls[0]).toMatchSnapshot();
        expect(execa.mock.calls[1]).toMatchSnapshot();
        expect(execa.mock.calls[2]).toMatchSnapshot();
    });
});
