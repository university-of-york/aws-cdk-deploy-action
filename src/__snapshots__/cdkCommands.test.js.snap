// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cdkCommands should call process and execa with correct arguments 1`] = `
Array [
  "npx",
  Array [
    "cdk",
    "bootstrap",
    "aws://account_id/eu-west-1",
    "--role-arn=arn:aws:iam::account_id:role/role_name",
    "--toolkit-stack-name=stack_name-cdk-toolkit",
  ],
  Object {
    "cwd": "some/path/",
  },
]
`;

exports[`cdkCommands should call process and execa with correct arguments 2`] = `
Array [
  "npx",
  Array [
    "cdk",
    "deploy",
    "--require-approval=never",
    "--role-arn=arn:aws:iam::account_id:role/role_name",
    "--toolkit-stack-name=stack_name-cdk-toolkit",
  ],
  Object {
    "cwd": "some/path/",
  },
]
`;
